src/turbolinks/history.coffee

```diff
@@ -20,16 +23,27 @@ class Turbolinks.History
 
   push: (location, restorationIdentifier) ->
     location = Turbolinks.Location.wrap(location)
+    locationStack.push(location)
     @update("push", location, restorationIdentifier)
 
   replace: (location, restorationIdentifier) ->
     location = Turbolinks.Location.wrap(location)
     @update("replace", location, restorationIdentifier)
 
+  back: (location, restorationIdentifier) ->
+    location = Turbolinks.Location.wrap(location)
+    
+    previousUrl = locationStack[locationStack.length - 2]
+    if previousUrl?.absoluteURL == location.absoluteURL
```

Comment:
Man, what a total bummer that you can't access the entries of the History API. Security, always getting in the way ;)